idaes/generic_models/properties/core/examples/tests/test_CO2_H2O_MEA_Ideal_VLE.py

```diff
@@ -0,0 +1,273 @@
+
+        assert isinstance(model.props[1].mole_frac_phase_comp, Var)
+        assert len(model.props[1].mole_frac_phase_comp) == 5
+
+        assert isinstance(model.props[1].phase_frac, Var)
+        assert len(model.props[1].phase_frac) == 2
+
+        assert isinstance(model.props[1].total_flow_balance, Constraint)
+        assert len(model.props[1].total_flow_balance) == 1
+
+        assert isinstance(model.props[1].component_flow_balances, Constraint)
+        assert len(model.props[1].component_flow_balances) == 3
+
+        assert isinstance(model.props[1].sum_mole_frac, Constraint)
+        assert len(model.props[1].sum_mole_frac) == 1
+
+        assert not hasattr(model.props[1], "sum_mole_frac_out")
+
+        assert isinstance(model.props[1].phase_fraction_constraint, Constraint)
+        assert len(model.props[1].phase_fraction_constraint) == 2
+
+        assert_units_consistent(model)
+
+    @pytest.mark.unit
+    def test_default_material_balance_type(self, model):
+        assert model.props[1].default_material_balance_type() == \
+            MaterialBalanceType.componentTotal
+
+    @pytest.mark.unit
+    def test_default_energy_balance_type(self, model):
+        assert model.props[1].default_energy_balance_type() == \
+            EnergyBalanceType.enthalpyTotal
+
+    @pytest.mark.unit
+    def test_get_material_flow_basis(self, model):
+        assert model.props[1].get_material_flow_basis() == \
+            MaterialFlowBasis.molar
+
+    @pytest.mark.unit
+    def test_define_state_vars(self, model):
+        sv = model.props[1].define_state_vars()
+
+        assert len(sv) == 4
+        for i in sv:
+            assert i in ["flow_mol",
+                         "mole_frac_comp",
+                         "temperature",
+                         "pressure"]
+
+    @pytest.mark.unit
+    def test_define_port_members(self, model):
+        sv = model.props[1].define_state_vars()
+
+        assert len(sv) == 4
+        for i in sv:
+            assert i in ["flow_mol",
+                         "mole_frac_comp",
+                         "temperature",
+                         "pressure"]
+
+    @pytest.mark.unit
+    def test_define_display_vars(self, model):
+        sv = model.props[1].define_display_vars()
+
+        assert len(sv) == 4
+        for i in sv:
+            assert i in ["Total Molar Flowrate",
+                         "Total Mole Fraction",
+                         "Temperature",
+                         "Pressure"]
+
+    @pytest.mark.unit
+    def test_dof(self, model):
+        # Fix state
+        model.props[1].flow_mol.fix(92.0817)
+        model.props[1].temperature.fix(380)
+        model.props[1].pressure.fix(183430)
+        model.props[1].mole_frac_comp["H2O"].fix(0.8787745)
+        model.props[1].mole_frac_comp["CO2"].fix(0.018288)
+        model.props[1].mole_frac_comp["MEA"].fix(0.1029375)
+
+        assert degrees_of_freedom(model.props[1]) == 0
+
+    @pytest.mark.initialize
```

Comment:
The `initialize`, `build` and `solver` marks are deprecated, so you can remove them.